### ISSUE TITLE
Explicit initialization of structures with boolean values using a default init "{0}" leads to type-system error

### ISSUE THREAD

**Issue by danielratiu** [mbeddr/mbeddr.core#1682]
struct s1 {
 boolean b1;
 boolean b2:
}

s1 v = {0} <-- type error (see screenshot)

![image](https://cloud.githubusercontent.com/assets/1437141/25312577/582c3586-281d-11e7-943c-821da34c12ce.png)

**Comment by andreasgrosche** [comment]
Regarding pure C, the initializer list initializes the named elements of the struct, starting with the first element.
The types of the elements in the initializer list have to match the types of the struct members in the same order.
The missing elements are initialized "implicitly the same as objects that have static storage duration" ("false" in this case).

E.g. the following code works for C (the first element is explicitely initialized with false and all other elements are implicitly initialized with the default value for static objects "false"):

```
struct s1 { 
  boolean b1; 
  boolean b2; 
};
                                             
s1 v = {false};  
```

Unfortunately, mbeddr complains, "wrong number of arguments".

For initialization in general, see also [http://stackoverflow.com/questions/11152160/initializing-a-struct-to-0](http://stackoverflow.com/questions/11152160/initializing-a-struct-to-0).

**Comment by danielratiu** [comment]
@andreasgrosche 
AFAIK each structure can be initialized (default) with "structVar = {0};"

We should allow this in mbeddr as well.

**Comment by coolya** [comment]
@danielratiu an empty initialiser list should work and give you the expected result.

**Comment by coolya** [comment]
@andreasgrosche yes C permits that but mbeddr currently not. But it would make good first contribution if you like 😉  We are always happy to see pull requests.
